cdc/sink/codec/avro.go

```diff
@@ -522,3 +525,69 @@ func (r *avroEncodeResult) toEnvelope() ([]byte, error) {
 	}
 	return buf.Bytes(), nil
 }
+
+type avroEventBatchEncoderBuilder struct {
+	credential *security.Credential
+	opts       map[string]string
+
+	keySchemaManager   *AvroSchemaManager
+	valueSchemaManager *AvroSchemaManager
+
+	initialized *atomic.Bool
+}
+
+func newAvroEventBatchEncoderBuilder(credential *security.Credential, opts map[string]string) EncoderBuilder {
+	return &avroEventBatchEncoderBuilder{
+		credential:  credential,
+		opts:        opts,
+		initialized: atomic.NewBool(false),
+	}
+}
+
+const (
+	keySchemaSuffix   = "-key"
+	valueSchemaSuffix = "-value"
+)
+
+// newSchemaManagers initialize the `keySchemaManager` and `valueSchemaManager` in an atomic way.
```

Comment:
I can't find any atomic relevant logic in this’ func. It make me a little bit confused.